uproot/write/objects/TH.py

```diff
@@ -83,6 +84,9 @@ def __init__(self, histogram):
         self.fields["_fBinEntries"] = numpy.array(self.fields["_fBinEntries"], dtype=">f8", copy=False)
         self.fields["_fBinSumw2"] = numpy.array(self.fields["_fBinSumw2"], dtype=">f8", copy=False)
 
+        #For _writeobjany
+        self._written = {}
```

Comment:
Is it absolutely necessary for the object to know that it has been written? We have a model in which

```python
file["something"] = histogram
```

makes a copy of the `histogram` that is unconnected to any other `histogram`—for instance,

```python
file["something_else"] = histogram
```

makes an entirely new, unconnected object in the file. For one thing, that file object has a different name, while the in-memory object has no name.

If `_writeobjany` wants to keep references to objects that have been written so that it doesn't write the same thing twice, well, we're writing the same thing twice to stay consistent with our "by value" rather than "by reference" model.

... Unless you can convince me otherwise that this model doesn't apply here?

Reply:
I completely overlooked this - the histogram object should not know that it has been written. (This would cause problems as soon as I started working on TTrees anyway)

But, there should be a dictionary somewhere holding the class names (and their corresponding indices) that have already been written. (Side note for future reference - the dictionary does not write the objects that the user wants to write the file - for eg. histograms, but it instead writes constituent parts such as TList and THashList).

I feel like moving the writeobjany (or as it is now called put_objany) to a different file/class and having the dictionary there as well, would be the correct course of action and I am currently trying to incorporate that.